server/src/test/java/org/opensearch/index/fielddata/ordinals/MultiOrdinalsTests.java

```diff
@@ -128,10 +128,10 @@ public int compare(OrdAndId o1, OrdAndId o2) {
                     assertThat((long) singleOrds.ordValue(), equalTo(docOrds.get(0)));
 
                     assertTrue(docs.advanceExact(docId));
+                    assertEquals(docOrds.size(), docs.docValueCount());
                     for (Long ord : docOrds) {
                         assertThat(docs.nextOrd(), equalTo(ord));
                     }
-                    assertEquals(SortedSetDocValues.NO_MORE_DOCS, docs.nextOrd());
```

Comment:
This will only be true in case of `SinglePackedOrdinals` whereas for `MultiOrdinals` the next method will still return NO_MORE_DOCS

Reply:
Eating my own words. 

We can change `MultiOrdinals$MultiDocs` to not return `NO_MORE_DOCS` and keep returning the last ord same as SingletonSSDV. 

```
        @Override
        public long nextOrd() throws IOException {
            if (currentOffset == currentEndOffset) {
                return SortedSetDocValues.NO_MORE_DOCS;
            } else {
                return ords.get(currentOffset++);
            }
        }
```

But then this will open up changing everything that depends on it which should anyways be done

Reply:
In general, IMO, we should assume that the behavior of `nextOrd()` when called more than `docValueCount()` times is undefined. 

The behavior of `MultiOrdinals$MultiDocs` is fine -- you're allowed to return `NO_MORE_DOCS`, I think, but callers should not assume that you do.

Reply:
@msfroh If we don't remove returning `SortedSetDocValues.NO_MORE_DOCS;` from `MultiOrdinals$MultiDocs` then any caller in future might rely on it and not check on the `docValueCount()`  

IMO we should enforce it by removing the return of `NO_MORE_DOCS` so that any such misses break the functionality.

Reply:
it is coming from `DocIdSetIterator.NO_MORE_DOCS` and has other useful purposes but shouldn't be used to compare max number of ordinals.

Reply:
Phew enforcing this means more changes :( 

Maybe we can take that up post 3.0 and deal with direct usages of `nextOrd()` for now.  Thoughts ?

Reply:
I don't think we need to change anything in `MultiOrdinals$MultiDocs` if `SortedSetDocValues`, which it implements, prohibits consumers to depend on `NO_MORE_DOCS` and clearly states to call `nextOrd()` only `docValueCount()` times.

Reply:
@rishabhmaurya If we don't remove returning `NO_MORE_DOCS` then we will be susceptible to this

> If we don't remove returning SortedSetDocValues.NO_MORE_DOCS; from MultiOrdinals$MultiDocs then any caller in future might rely on it and not check on the docValueCount()

Reply:
@expani I slightly disagree here as in general consumer of an API should comply with the API contract and not care about implementation's logic and make assumptions.

Reply:
@rishabhmaurya I meant places like [MultiOrdinals$MultiValues.nextOrd()](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/index/fielddata/ordinals/MultiOrdinals.java#L221-L222) don't comply with the API contract of SortedSetDocValues by returning `NO_MORE_DOCS` instead of the last accessed ordinal. 

So, shouldn't that be changed to return the last accessed ordinal ?

Reply:
@expani I think it is complying with `nextOrd()` the contract as contract isn't saying anywhere to return the last accessed ordinal.  

```
  /**
   * Returns the next ordinal for the current document. It is illegal to call this method after
   * {@link #advanceExact(int)} returned {@code false}. It is illegal to call this more than {@link
   * #docValueCount()} times for the currently-positioned doc.
   *
   * @return next ordinal for the document. ordinals are dense, start at 0, then increment by 1 for
   *     the next value in sorted order.
   */
  public abstract long nextOrd() throws IOException;
```